### PR TITLE
core: fix simd emulator code

### PR DESCRIPTION
**Pull request by alalek**
**Merge with contrib**: https://github.com/opencv/opencv_contrib/pull/2403

Usage:

```
cmake -DOPENCV_EXTRA_FLAGS="-DCV_FORCE_SIMD128_CPP=1" ...
```

<cut/>

```
force_builders=Custom
buildworker:Custom=linux-1
build_image:Custom=simd-emulator

#test_timeout:Linux x64 Debug=1200
#test_maxtime:Linux x64 Debug=14400
```

### CODE REVIEW

File: modules/core/include/opencv2/core/hal/intrin_cpp.hpp [hunk 1]
```diff
@@ -1659,22 +1773,19 @@ Pointer __should__ be aligned by 16-byte boundary. */
 template<typename _Tp, int n>
 inline void v_store_aligned(_Tp* ptr, const v_reg<_Tp, n>& a)
 {
-    for( int i = 0; i < n; i++ )
-        ptr[i] = a.s[i];
+    v_store(ptr, a);
```
Comment (terfendail) [inline modules/core/include/opencv2/core/hal/intrin_cpp.hpp]: May be it make sense to add alignment check here?

File: modules/core/src/arithm.simd.hpp [hunk 2]
```diff
@@ -1593,7 +1593,7 @@ struct op_div_f
 {
     static inline Tvec r(const Tvec& a, const Tvec& b)
     {
-        const Tvec v_zero = Tvec();
+        const Tvec v_zero = vx_setall<typename Tvec::lane_type>(0);
```
Comment (terfendail) [inline modules/core/src/arithm.simd.hpp]: IMO it make sense to add vx_setzero template as well and use it here instead

Comment (alalek) [reply modules/core/src/arithm.simd.hpp]: Will be added as a separate PR

File: modules/core/src/mathfuncs_core.simd.hpp [hunk 3]
```diff
@@ -725,7 +725,7 @@ void log32f( const float *_x, float *y, int n )
 
         yf0 = v_fma(v_cvt_f32(yi0), vln2, yf0);
 
-        v_float32 delta = v_reinterpret_as_f32(h0 == vx_setall_s32(510)) & vshift;
+        v_float32 delta = v_select(v_reinterpret_as_f32(h0 == vx_setall_s32(510)), vshift, vx_setall<float>(0));
```
Comment (terfendail) [inline modules/core/src/mathfuncs_core.simd.hpp]: There is a possibility that this version will be slower on SSE2 and SSE3. However I think the line could be optimized by a compiler.

**Pull request merged**